main.cpp

```diff
@@ -1,109 +1,228 @@
+/* Sockets Example
+ * Copyright (c) 2016-2020 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
 #include "mbed.h"
 
-// Network interface
-NetworkInterface *net;
+class SocketDemo {
+    static const size_t MAX_NUMBER_OF_ACCESS_POINTS = 10;
+    static const size_t MAX_MESSAGE_RECEIVED_LENGTH = 100;
 
-// Socket demo
-int main() {
-    int remaining;
-    int rcount;
-    char *p;
-    char buffer[256];
-    nsapi_size_or_error_t result;
+public:
+    SocketDemo()
+    {
+        _net = NetworkInterface::get_default_instance();
+    }
 
-    // Bring up the ethernet interface
-    printf("Mbed OS Socket example\n");
+    ~SocketDemo()
+    {
+        _net->disconnect();
+    }
 
-#ifdef MBED_MAJOR_VERSION
-    printf("Mbed OS version: %d.%d.%d\n\n", MBED_MAJOR_VERSION, MBED_MINOR_VERSION, MBED_PATCH_VERSION);
-#endif
+    void run()
+    {
+        if (!_net) {
+            printf("Error! No network interface found.\r\n");
+            return;
+        }
 
-    net = NetworkInterface::get_default_instance();
+        /* if we're using a wifi interface run a quick scan */
```

Comment:
Is it required to scan when we want to connect to the hardcoded access point ? It is important to demonstrate the scan feature but if scan is not required to connect could you detail it in the comment.